Entities/Activity.cpp

```diff
@@ -0,0 +1,1136 @@
+		if (m_ControlledActor[player] && g_MovableMan.IsActor(m_ControlledActor[player])) {
+			// Set AI controller active to the guy we're switching from
+			m_ControlledActor[player]->SetControllerMode(Controller::CIM_AI);
+			m_ControlledActor[player]->GetController()->SetDisabled(false);
+			prevActor = m_ControlledActor[player];
+		} else {
+			m_ControlledActor[player] = 0;
+		}
+
+		// Get previous Actor before the current one.
+		m_ControlledActor[player] = g_MovableMan.GetPrevTeamActor(team, m_ControlledActor[player]);
+
+		// If it's the skip guy, then skip him OR if it's a non-brain guy controlled by another player on the same team, then skip him too OR if it's a brain of another player
+		Actor *start = m_ControlledActor[player];
+		while ((skip && skip == m_ControlledActor[player]) || (m_ControlledActor[player] && m_ControlledActor[player] != m_Brain[player] && m_ControlledActor[player]->IsPlayerControlled()) || (IsOtherPlayerBrain(m_ControlledActor[player], player))) {
+			// Check previous in line
+			m_ControlledActor[player] = g_MovableMan.GetPrevTeamActor(team, m_ControlledActor[player]);
+
+			// Looped around the whole actor chain, and couldn't find an available actor, so switch back to the original
+			if (m_ControlledActor[player] == prevActor) {
+				g_GUISound.UserErrorSound()->Play();
+				m_ControlledActor[player] = prevActor;
+				break;
+			}
+			// If we're gone around without even finding the one we started with (he might be dead), then try to switch to the brain, and if not that, then 0
+			if (m_ControlledActor[player] == start) {
+				if (m_Brain[player]) {
+					m_ControlledActor[player] = m_Brain[player];
+				} else if (g_MovableMan.IsActor(start)) {
+					m_ControlledActor[player] = start;
+				} else {
+					m_ControlledActor[player] = 0;
+				}
+				break;
+			}
+		}
+
+		// Now set the controller of the actor we switched to, and play switching sound effects
+		if (m_ControlledActor[player]) {
+			m_ControlledActor[player]->SetControllerMode(Controller::CIM_PLAYER, player);
+
+			// Play actor switching sound effects
+			(m_ControlledActor[player] == m_Brain[player]) ? g_GUISound.BrainSwitchSound()->Play() : g_GUISound.ActorSwitchSound()->Play();
+			// Only play air swoosh if actors are out of sight of each other
+			if (prevActor && Vector(prevActor->GetPos() - m_ControlledActor[player]->GetPos()).GetMagnitude() > g_FrameMan.GetResX() / 2) { g_GUISound.CameraTravelSound()->Play(); }			
+
+			ReassignSquadLeader(player, team);
+		}
+
+		// Follow the new guy normally
+		m_ViewState[player] = NORMAL;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Activity::EnteredOrbit(Actor *orbitedCraft) {
+		if (!orbitedCraft) {
+			return;
+		}
+		int team = orbitedCraft->GetTeam();
+
+		// Find out cost multipliers for metagame, or we end up AI's earning money when their craft return if their nativeCostMultiplier is too low
```

Comment:
No need for any of the comments in this method, except maybe the last one.

I'd say maybe pull out the meta cost multiplier stuff into a separate private method though.